libs/base/src/lib/context/PricesProvider.tsx

```diff
@@ -1,5 +1,111 @@
+import { createContext, useContext, useEffect } from 'react'
+
 import { createPricesContext } from '@apps/hooks'
+import { useFetchState } from '@apps/hooks'
+import { usePrevious } from 'react-use'
+import { useBlockNumber } from 'wagmi'
+
+import { ChainIds, useNetwork } from './NetworkProvider'
+
+import type { FetchState } from '@apps/types'
+import type { FC } from 'react'
+
+interface NetworkPrices {
+  nativeToken?: number
+  gas?: {
+    slow: number
+    standard: number
+    fast: number
+    instant: number
+  }
+}
+
+interface GasPrice {
+  standard: number
+  fast: number
+  slow: number
+  instant: number
+}
+
+interface MyCryptoGas {
+  safeLow: number
+  standard: number
+  fast: number
+  fastest: number
+}
+
+interface MaticMainGas {
+  safeLow: number
+  standard: number
+  fast: number
+  fastest: number
+  blockTime: number
+  blockNumber: number
+}
 
 const [useFetchPriceCtx, PricesProvider] = createPricesContext()
 
 export { useFetchPriceCtx, PricesProvider }
+
+const networkPricesCtx = createContext<FetchState<NetworkPrices>>(null as never)
+
+export const NetworkPricesProvider: FC = ({ children }) => {
+  const network = useNetwork()
+  const { data: block } = useBlockNumber({ watch: true })
+  const previousBlock = usePrevious(block)
+
+  const [networkPrices, setNetworkPrices] = useFetchState<NetworkPrices>({})
+
+  useEffect(() => {
+    const fetchPrices = async () => {
+      setNetworkPrices.fetching()
+
+      let gas: GasPrice
+
+      // eth mainnet
+      if (network.chainId === ChainIds.EthereumMainnet) {
+        const gasStationResponse = await fetch(network.gasStationEndpoint)
+        const gasRes: MyCryptoGas = await gasStationResponse.json()
+        gas = {
+          standard: gasRes.standard,
+          fast: gasRes.fast,
+          slow: gasRes.safeLow,
+          instant: gasRes.fastest,
+        }
+        // eth testnet
+      } else if ([ChainIds.EthereumGoerli, ChainIds.EthereumKovan, ChainIds.EthereumRopsten].includes(network.chainId)) {
+        // Testnets should use low gas
+        gas = {
+          standard: 3,
+          fast: 3,
+          slow: 3,
+          instant: 3,
+        }
+        // Matic Mainnet + Mumbai
+      } else {
+        const gasStationResponse = await fetch(network.gasStationEndpoint)
+        const gasRes: MaticMainGas = await gasStationResponse.json()
+        gas = {
+          standard: Math.min(30, gasRes.standard),
```

Comment:
Why the min 30 on polygon ?

Reply:
tbh, I don't know... this is copied from previous version

Reply:
This is a limitation by polygon. They implemented that the min gas is 30 gwei. If you try to send tx lower than that it might be stuck forever in the mempool or the RPC will just reject it.